src/drop.ts

```diff
@@ -35,8 +39,10 @@ export function drop(...args: ReadonlyArray<unknown>): unknown {
   return purry(dropImplementation, args, lazyImplementation);
 }
 
-const dropImplementation = <T>(array: ReadonlyArray<T>, n: number): Array<T> =>
-  n < 0 ? [...array] : array.slice(n);
+const dropImplementation = <const T extends Array<unknown>, N extends number>(
+  array: T,
+  n: N,
+): Drop<T, N> => (n < 0 ? [...array] : array.slice(n)) as never;
```

Comment:
Don't use `as never`. It's not better than disabling the typescript error via `// @ts-expect error`, which makes the typing issue more obvious.

anyway, because this is the implementation and it's not always easy to make typescript aware of complex types return the trivial type.
```suggestion
const dropImplementation = <T extends IterableContainer>(
  array: T,
  n: number,
): Array<T[number]> => (n < 0 ? [...array] : array.slice(n));
```

Reply:
Sure
Done